### DISCUSSION TITLE
Property Research (including Player Housing) - WIP

### DISCUSSION THREAD

**Discussion by jasper-one** [EveronLife/EveronLife#174] [General]
## **Overview**

This discussion contains some initial research into both the residential and commercial properties of the Everon map and what the current assets might mean in the context of creating Everon Life.

There are a number of key themes that will need to be considered that are different than what was required in Arma 3 due to the added level of detail of these properties and the more widely varied types of properties in the game.

#### **Residential**

**Property Types**

Different from Arma 3, different properties have very real differences in Everon. Some properties come with barely more than a bed, while others can have 5+ utilities (explained below). This means property progression could be a very core aspect of the game. For example, a player may start living in a Fishermen's Shed and eventually own multiple Commercial & Industrial properties, owning many properties in between. 

**Utilities**

It seems the people of Everon are highly self-sufficient. Many of the village houses come complete with garden beds, workshops, water pumps, chicken coups, and other utilities. This represents a unique opportunity to make the property an even more key part of Everon Life as each property has very unique capabilities. Properties hypothetically are the primary means of gathering and crafting in the mid and end-game.

**Town Houses**

In Everon there is a concept of a true Town House where there are multiple units/apartments in the same building. These buildings also have shared/common areas such as bathrooms, workshops, and attics. Meaning that for the first time, multiple players may have to share certain aspects of a property (e.g. even the front door).

#### **Commercial**

Everon also has many more variations of commercial buildings. Rather than be simply limited to general stores and gas stations, Everon has over 10 different types of commercial property that can be found around the map. Again providing Everon Life with a lot of opportunities. 

#### **Industrial**
Large industrial properties with various capabilities are also across the map, such as Ports, Warehouses, and Factories. 

### **Fishermen Sheds**
_A great starter property_

There are 31 Fishermen's Sheds around the map, found on the coastline in fishing villages. There was no observed additional utilities near these properties (e.g. no areas for crops, workshops, etc).
![image](https://user-images.githubusercontent.com/106525617/196912818-92879929-963f-42ca-928b-caf3af2e9965.png)

Inside you can generally find a small storage chest, a stove, a seat, and some wood. 

**In-Game Example:**
- [ ] Add Workbench Links

**Pefabs:**

Prefabs/Structures/Infrastructure/Naval/Sheds/FishermanShed_01_base.et \
Prefabs/Structures/Infrastructure/Naval/Sheds/FishermanShed_02_base.et

### **Town Houses** 
_Standard Mid-Game Player Housing_
![image](https://user-images.githubusercontent.com/106525617/196912961-852cfccb-52f8-47e7-a156-a49930df4904.png)

Found in the main towns/cities these buildings can contain up to ~4 units and 2 shared bathrooms alongside other shared spaces such as a workshop and an attic. These properties are much more limited than “Village houses” as none of the three variations provides a garage. Rarely these buildings will have a shared garden area.  

The number of buildings and units is: 
* 7 x Type 2I01_v2 with 2 Units Each
* 13 x Type 2I01 with 2 Units Each
* 11 x Type 2I02 with 2 Units Each
* 8x Type 2I03 with upto 4 Units Each (depending on how we want to split it)
* 1x Type 2I03_v2 upto 4 Units Each 

This means the amount of units totals between 94 and 80 depending on what is considered an apartment for Type Type 2I03 buildings. 

**In-Game Example:**
- [ ] Add Workbench Links

**Prefabs:**
Prefabs/Structures/Houses/Town/House_Town_E_2I01_Base.et
Prefabs/Structures/Houses/Town/House_Town_E_2I02_Base.et
Prefabs/Structures/Houses/Town/House_Town_E_2I03_Base.et

### Village Houses
_Late Game Housing_
![image](https://user-images.githubusercontent.com/106525617/196913054-35d50da7-ac5e-45a2-831e-1a49dff89d30.png)

1. House
2. Garage
3. Workshop/Barn
4. Garden Bed
5. Chicken Coop
6. Water Pump
7. Storage Shed

**Additional Example**
![image](https://user-images.githubusercontent.com/106525617/196913143-baa4e5d0-2e9b-4863-a41f-409540977f58.png)

1. House
2. Garage
3. Three-Door Barn
4. Garden Bed
5. Glass House
6. Workshop
7. Chicken Coop 

### Mountain Houses
_End Game Housing_
These are rural properties observed out of town. The standard version has a garage while the short version does not.
![image](https://user-images.githubusercontent.com/106525617/196913232-3141586f-2c6b-4d86-a050-3af7c703fbd4.png)
![image](https://user-images.githubusercontent.com/106525617/196913339-a073238c-7a7e-4014-acf7-835f9bf06d03.png)

The majority of these properties have little if any additional utilities. The above is one of the best examples of a Mountain House that does have utilities.

The utilities for this property are: 
1. House
2. Garage
3. Vegatable Bed
4. Glass House
5. Shed
6. Chicken Coop
7. Water Pump 

The number of mountain houses is: 
* 40 x 1I01_short
* 26 x 1I01

**Prefabs:**
Prefabs/Structures/Houses/Village/House_Mountain_E_1I01_short.et
Prefabs/Structures/Houses/Village/House_Mountain_E_1I01.et

### **Boat Sheds**
_Property to store watercraft_
![image](https://user-images.githubusercontent.com/106525617/196913574-48dd7fb6-8772-4dc1-b8bc-6dc472c1be93.png)

### **Stall**
![image](https://user-images.githubusercontent.com/106525617/196913588-25344918-f534-4646-be4e-b97be05c844e.png)

**Shop House**
![image](https://user-images.githubusercontent.com/106525617/196913633-2aa69e3c-f0ed-426f-a5de-40fc1a0dd7ca.png)
One-bed studio apartment on top of the store with an attic

### **Barn**
![image](https://user-images.githubusercontent.com/106525617/196914004-895120e5-af1b-4236-9c51-110363fe5e4d.png)

### **Barn House**
![image](https://user-images.githubusercontent.com/106525617/196914047-14d2af05-2e6f-4f13-a496-b53dc6de700f.png)

### **Mountain Hotel**
![image](https://user-images.githubusercontent.com/106525617/196914104-81045796-055b-4082-b04c-9687a54be1f7.png)
_Red_
![image](https://user-images.githubusercontent.com/106525617/196914140-13c471b5-001f-4ec4-aa00-9fe53039fc7e.png)
_Green_

### **Restaurant**
![image](https://user-images.githubusercontent.com/106525617/196914176-bf15e917-5fb3-4381-a375-8ad19305278c.png)
![image](https://user-images.githubusercontent.com/106525617/196914199-5bb088c0-0659-4041-9426-1361a41fb010.png)

### **General Store**
![image](https://user-images.githubusercontent.com/106525617/196914239-f6b1d345-de1c-452a-aae0-924073c8399c.png)

## Industrial
![image](https://user-images.githubusercontent.com/106525617/196914319-5fd01604-8a7e-4ea4-b870-d5c43d9b185d.png)

## Government

**Fire Station **
![image](https://user-images.githubusercontent.com/106525617/196914337-8bf1cf5f-f4d4-49b4-a6da-ad1c21a73f12.png)

### **Water Pump**
![image](https://user-images.githubusercontent.com/106525617/196914384-8a446f44-960a-4bea-8d11-d7d3446e9fb2.png)

### **Bee Hives**
![image](https://user-images.githubusercontent.com/106525617/196914403-576ca806-8321-4165-ac53-527561b16d0e.png)

### **Glass House**
![image](https://user-images.githubusercontent.com/106525617/196914423-51b09e27-91a6-40f0-bf99-f2e22e69a48f.png)

## Other Notes

* Multiple people (not just the owner) can use it as a spawn point. The number of beds in the property could be used as the maximum number of characters that can use as a spawn point.

**Comment by thebonbon** [comment]
> Multiple people (not just the owner) can use it as a spawn point. The number of beds in the property could be used as the maximum number of characters that can use as a spawn point.
>
Great idea, like in Rust you need to have a dedicated bed for everyone that wants to spawn there.
